service/src/main/java/uk/gov/hmcts/reform/fpl/service/validators/ManageLegalRepresentativesValidator.java

```diff
@@ -0,0 +1,102 @@
+import uk.gov.hmcts.reform.fpl.service.OrganisationService;
+
+import java.text.MessageFormat;
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import static java.util.Objects.isNull;
+import static org.apache.commons.lang3.StringUtils.isEmpty;
+
+@Component
+@Slf4j
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class ManageLegalRepresentativesValidator {
+
+    private static final String VALIDATION_SUFFIX = "for Legal representative";
+
+    private final OrganisationService organisationService;
+
+    public List<String> validate(List<Element<LegalRepresentative>> legalRepresentatives) {
+        return Streams.mapWithIndex(
+            Optional.ofNullable(legalRepresentatives).orElse(Lists.emptyList()).stream(),
+            (legalRepresentativeElement, idx) ->
+                performBasicValidation(legalRepresentativeElement.getValue(), idx, legalRepresentatives.size())
+        ).flatMap(Collection::stream).collect(Collectors.toList());
+    }
+
+    private List<String> performBasicValidation(LegalRepresentative legalRepresentative, long currentRepresentativeIdx,
+                                                int sizeOfRepresentatives) {
+        List<String> validationErrors = Lists.newArrayList();
+        if (isEmpty(legalRepresentative.getFullName())) {
+            validationErrors.add(validationMessage("Enter a full name",
+                currentRepresentativeIdx, sizeOfRepresentatives));
+        }
+
+        if (isNull(legalRepresentative.getRole())) {
+            validationErrors.add(validationMessage("Select a role",
+                currentRepresentativeIdx, sizeOfRepresentatives));
+        }
+
+        if (isEmpty(legalRepresentative.getOrganisation())) {
+            validationErrors.add(validationMessage("Enter an organisation",
+                currentRepresentativeIdx,
+                sizeOfRepresentatives));
+        }
+
+        if (isEmpty(legalRepresentative.getEmail())) {
+            validationErrors.add(validationMessage("Enter an email address",
+                currentRepresentativeIdx,
+                sizeOfRepresentatives));
+        } else {
+            Optional<String> userId = organisationService.findUserByEmail(legalRepresentative.getEmail());
+            if (userId.isEmpty()) {
+                validationErrors.add(
+                    validationMessageForInvalidEmail(currentRepresentativeIdx, sizeOfRepresentatives));
+            }
+        }
+
+        if (isEmpty(legalRepresentative.getTelephoneNumber())) {
+            validationErrors.add(validationMessage("Enter a phone number",
+                currentRepresentativeIdx,
+                sizeOfRepresentatives)
+            );
+        }
+
+        return validationErrors;
+
+    }
+
+    private String validationMessage(String prefix, long currentRepresentativeIdx, int sizeOfRepresentatives) {
+        return MessageFormat.format("{0} {1}{2}",
```

Comment:
minor, in most places we use String.format

Reply:
intellij got too intelligent this time